src/pages/stacks-blockchain/integrate-stacking.md

```diff
@@ -223,114 +168,87 @@ Users can select how many cycles they would like to participate in. To help with
-      `0x${serializeCV(uintCV(poxInfo.reward_cycle_id)).toString('hex')}`,
-      `0x${serializeCV(uintCV(numberOfCycles)).toString('hex')}`,
-    ],
-  },
-});
+// user supplied parameters
+let btcAddress = '1Xik14zRm29UsyS6DjhYg4iZeZqsDa8D3';
+let cycles = 3;
 
-const response = cvToString(deserializeCV(Buffer.from(isEligible.result.slice(2), 'hex')));
+const stackingEligibility = await client.canStack({ btcAddress, cycles });
 
-if (response.startsWith(`(err `)) {
-  // user cannot participate in stacking
-  // error codes: https://github.com/blockstack/stacks-blockchain/blob/master/src/chainstate/stacks/boot/pox.clar#L2
-  console.log({ isEligible: false, errorCode: response }));
-  return;
-}
-// success
-console.log({ isEligible: true });
+// stackingEligibility:
+// {
+//   eligible: false,
+//   reason: 'ERR_STACKING_INVALID_LOCK_PERIOD',
+// }
 ```
 
-If the user is eligible, the stacking action should be enabled on the UI. If not, the respective error message should be shown to the user.
+-> Note that the eligibility check assumes the user will be stacking the maximum balance available in the account.
 
--> For more information on this read-only API call, please review the [API references](https://blockstack.github.io/stacks-blockchain-api/#operation/call_read_only_function)
+-> The eligibility check is a read-only function call to the PoX smart contract which does not require broadcasting a transaction
 
-## Step 5: Add stacking action
+If the user is eligible, the stacking action should be enabled on the UI. If not, the respective error message should be shown to the user.
 
-Next, the Stacking action should be implemented. Once the user confirms the action, a new transaction needs to be broadcasted to the network:
+## Step 5: Lock STX to stack
 
-```js
-const tx = new TransactionsApi(apiConfig);
+Next, the Stacking action should be executed.
 
-const [contractAddress, contractName] = poxInfo.contract_id.split('.');
-const network = new StacksTestnet();
-
-const txOptions = {
-  contractAddress,
-  contractName,
-  functionName: 'stack-stx',
-  functionArgs: [
-    uintCV(microstacksoLockup),
-    tupleCV({
-      hashbytes,
-      version,
-    }),
-    uintCV(coreInfo.burn_block_height),
-    uintCV(numberOfCycles),
-  ],
-  senderKey: privateKey.data.toString('hex'),
-  validateWithAbi: true,
-  network,
-};
-
-const transaction = await makeContractCall(txOptions);
-
-const contractCall = await broadcastTransaction(transaction, network);
-
-// this will return a new transaction ID
-console.log(contractCall);
+```js
+// specify the reward BTC address
+const btcAddress = '1Xik14zRm29UsyS6DjhYg4iZeZqsDa8D3';
+
+// set the amount to lock in microstacks
+const amountMicroStx = new BN(100000000000);
+
+// set the number of cycles to lock
+const cycles = 10;
+
+// set the burnchain (BTC) block for stacking lock to start
+// you can find the current burnchain block height from coreInfo above
+const burnBlockHeight = 2136;
```

Comment:
let's use the burn height returned from the pox info here to make the connection for the reader